Source/Scene/Points3DTileContent.js

```diff
@@ -164,159 +196,366 @@ define([
+        }
+
+        var shaderProgram = ShaderProgram.fromCache({
+            context : context,
+            vertexShaderSource : vs,
+            fragmentShaderSource : fs,
+            attributeLocations : attributeLocations
         });
 
-        this._primitive = primitive;
-        this.state = Cesium3DTileContentState.PROCESSING;
-        this.contentReadyToProcessPromise.resolve(this);
+        content._opaqueRenderState = RenderState.fromCache({
+            depthTest : {
+                enabled : true
+            }
+        });
 
-        var that = this;
+        content._translucentRenderState = RenderState.fromCache({
+            depthTest : {
+                enabled : true
+            },
+            depthMask : false,
+            blending : BlendingState.ALPHA_BLEND
+        });
 
-        when(primitive.readyPromise).then(function(primitive) {
-            that.state = Cesium3DTileContentState.READY;
-            that.readyPromise.resolve(that);
-        }).otherwise(function(error) {
-            that.state = Cesium3DTileContentState.FAILED;
-            that.readyPromise.reject(error);
+        content._drawCommand = new DrawCommand({
+            boundingVolume : content._tile.contentBoundingVolume.boundingSphere,
+            cull : false, // Already culled by 3D tiles
+            modelMatrix : new Matrix4(),
+            primitiveType : PrimitiveType.POINTS,
+            vertexArray : vertexArray,
+            count : pointsLength,
+            shaderProgram : shaderProgram,
+            uniformMap : uniformMap,
+            renderState : isTranslucent ? content._translucentRenderState : content._opaqueRenderState,
+            pass : isTranslucent ? Pass.TRANSLUCENT : Pass.OPAQUE,
+            owner : content
         });
-    };
+
+        content.state = Cesium3DTileContentState.READY;
+        content.readyPromise.resolve(content);
+        content._featureTableResources = undefined; // Unload
+    }
 
     /**
      * Part of the {@link Cesium3DTileContent} interface.
      */
     Points3DTileContent.prototype.applyDebugSettings = function(enabled, color) {
-        color = enabled ? color : this._constantColor;
-        this._primitive.appearance.uniforms.highlightColor = color;
+        this._highlightColor = enabled ? color : this._constantColor;
     };
 
+    var scratchMatrix = new Matrix3();
+
     /**
      * Part of the {@link Cesium3DTileContent} interface.
      */
     Points3DTileContent.prototype.update = function(tileset, frameState) {
-        // In the PROCESSING state we may be calling update() to move forward
-        // the content's resource loading.  In the READY state, it will
-        // actually generate commands.
-        this._primitive.update(frameState);
+        if (!defined(this._drawCommand)) {
+            createResources(this, frameState);
+        }
+
+        // Update the model matrix
+        var boundingSphere = this._tile.contentBoundingVolume.boundingSphere;
```

Comment:
Is it easy/worth-it to have a dirty flag to avoid burning this bit of CPU?  A full 4x4 matrix compare would not be worth it.

I only ask because this is per tile CPU only head, per tileset would not be an issue.